test/e2e/e2ethanos/services.go

```diff
@@ -814,37 +822,50 @@ http {
 // NewMinio returns minio server, used as a local replacement for S3.
 // TODO(@matej-g): This is a temporary workaround for https://github.com/efficientgo/e2e/issues/11;
 // after this is addresses fixed all calls should be replaced with e2edb.NewMinio.
-func NewMinio(env e2e.Environment, name, bktName string) *e2e.InstrumentedRunnable {
+func NewMinio(env e2e.Environment, name, bktName string) (*e2e.InstrumentedRunnable, error) {
 	image := "minio/minio:RELEASE.2019-12-30T05-45-39Z"
 	minioKESGithubContent := "https://raw.githubusercontent.com/minio/kes/master"
 	commands := []string{
 		"curl -sSL --tlsv1.2 -O '%s/root.key'	-O '%s/root.cert'",
-		"mkdir -p /data/%s && minio server --address :%v --quiet /data",
+		"mkdir -p /data/%s && minio server --certs-dir /shared/data/certs --address :%v --quiet /data",
+	}
+
+	if err := os.MkdirAll(filepath.Join(env.SharedDir(), "data", "certs", "CAs"), 0750); err != nil {
+		return nil, errors.Wrap(err, "create certs dir")
+	}
+
+	if err := genCerts(
+		filepath.Join(env.SharedDir(), "data", "certs", "public.crt"),
+		filepath.Join(env.SharedDir(), "data", "certs", "private.key"),
+		filepath.Join(env.SharedDir(), "data", "certs", "CAs", "ca.crt"),
+		env.Name()+"-"+name); err != nil {
+		return nil, errors.Wrap(err, "fail to generate certs")
 	}
 
 	return e2e.NewInstrumentedRunnable(
 		env,
 		name,
-		map[string]int{"http": 8090},
-		"http").Init(
+		map[string]int{"https": 8090},
+		"https").Init(
 		e2e.StartOptions{
 			Image: image,
 			// Create the required bucket before starting minio.
-			Command:   e2e.NewCommandWithoutEntrypoint("sh", "-c", fmt.Sprintf(strings.Join(commands, " && "), minioKESGithubContent, minioKESGithubContent, bktName, 8090)),
-			Readiness: e2e.NewHTTPReadinessProbe("http", "/minio/health/ready", 200, 200),
+			Command: e2e.NewCommandWithoutEntrypoint("sh", "-c", fmt.Sprintf(strings.Join(commands, " && "), minioKESGithubContent, minioKESGithubContent, bktName, 8090)),
+			//TODO(@clyang82): This is a temporary workaround for https://github.com/efficientgo/e2e/issues/9
+			//Readiness: e2e.NewHTTPReadinessProbe("http", "/minio/health/ready", 200, 200),
```

Comment:
I'm wondering if this could not increase flakiness somewhere down the road, since we'll be assuming `minio` is ready straight away :thinking:. Could we try to do the readiness probe with the workaround I described in that commented issue?

Reply:
Do you mean try to do the readiness probe with the workaround here? I think we should support this feature in `efficientgo/e2e` and then update here.

Reply:
I'm not sure how soon that will be fixed in the upstream, but the flakiness could be immediate. I would agree with either solution (workaround or fix it upstream) as long as it can be done ASAP.

Reply:
@matej-g please look at the fix in upstream - https://github.com/efficientgo/e2e/pull/19
I also have a workaround for the readiness probe so that we can merge it this PR firstly. I can open another PR since the upstream fix is acceptable.

Reply:
Awesome! Thanks a ton for addressing my concern, this is looking good now 👍. We can replace it with proper option once it lands in the upstream.